io/driver/driver_parameter.py

```diff
@@ -0,0 +1,155 @@
+        self.error_modules = []
+        self.mod_list = []
+        self.uname = linux_modules.platform.uname()[2]
+        if self.built_in_module(self.module) is True:
+            self.cancel("Module %s is Built-in Skipping " %
+                        self.module)
+        if self.param_check() is False:
+            self.cancel("Param %s is not Valid for Module %s" %
+                        (self.param_name, self.module))
+
+    def built_in_module(self, module):
+        """
+        checking whether the given module is built_in module or not
+        """
+        path = "/lib/modules/%s/modules.builtin" % self.uname
+        for each in genio.read_all_lines(path):
+            out = process.getoutput(each.split('/')[-1])
+            if module == out.split('.'[0]):
+                return True
+            return False
+
+    def sysfs_value_check(self):
+        '''
+        Checks if sysfs value matches to test value, Returns True if yes.
+        Returns False otherwise.
+        '''
+        value_check = process.system_output('/usr/bin/cat \
+                               /sys/module/%s/parameters/%s' %
+                                            (self.module, self.param_name
+                                             )).decode('utf-8')
+        if self.param_value not in value_check:
+            return False
+        return True
+
+    def param_check(self):
+        '''
+        Checks if Param is available for the Module, Returns True if yes.
+        Returns False otherwise.
+        '''
+        value_check = process.system_output('/usr/sbin/modinfo -p %s '
+                                            % self.module).decode('utf-8')
+        if self.param_name not in value_check:
+            return False
+        return True
+
+    def module_load_unload(self, mod1):
+        """
+        Unloading and loading the given module
+        """
+        if linux_modules.module_is_loaded(mod1) is False:
+            linux_modules.load_module(mod1)
+            time.sleep(self.load_unload_sleep_time)
+        sub_mod = linux_modules.get_submodules(mod1)
+        if sub_mod:
+            for mod in sub_mod.split(' '):
+                linux_modules.unload_module(mod)
+                if linux_modules.module_is_loaded(mod) is True:
+                    self.error_modules.append(mod)
+                    break
+        if linux_modules.unload_module(mod1) is False:
+            self.fail("Unloading Module %s failed" % mod1)
+        time.sleep(self.load_unload_sleep_time)
+        cmd = "%s %s=%s" % (mod1, self.param_name, self.param_value)
+        if linux_modules.load_module(cmd) is False:
+            self.fail("Param %s = Value %s Failed for Module %s" %
+                      (self.param_name, self.param_value, mod1))
+        if self.sysfs_chk:
+            if self.sysfs_value_check() is False:
+                self.fail("Sysfs check failed ")
+        if not HostInfo.ping_check(self, self.ifaces, self.peer, '1000',
```

Comment:
@manvanthar we should set IP Addr here as well, using `configure_network.set_ip`

Reply:
@manvanthar is this resolved ?

Reply:
@narasimhan-v   my comments are missing :(

configure_network.set_ip is not needed before ping_check.   As it is covered in Setup.  setting ifcfg is an one time activity which is done in Setup.  

After which , module load and unload just does ifup to pick the IP address.

Reply:
Ah sure, thanks @manvanthar